src/CameraSensor.cc

```diff
@@ -403,12 +409,12 @@ bool CameraSensor::Update(const std::chrono::steady_clock::duration &_now)
 
   if (!this->dataPtr->pub.HasConnections() &&
       this->dataPtr->imageEvent.ConnectionCount() <= 0 &&
-      !this->dataPtr->saveImage)
+      !this->dataPtr->saveImage && !this->dataPtr->infoPub.HasConnections())
```

Comment:
This condition could be simplified only if the return statement on line would be removed.